order-book/src/test/java/com/market/banica/order/book/model/ItemMarketTest.java

```diff
@@ -14,24 +21,94 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
-class ItemMarketTest {
+public class ItemMarketTest {
     private static final String EGGS_ITEM_NAME = "eggs";
     private static final String RICE_ITEM_NAME = "rice";
     private static final String MEAT_ITEM_NAME = "meat";
+    private static final String CHEESE_ITEM_NAME = "cheese";
 
     private static final String ALL_ITEMS_FIELD = "allItems";
+    private static final String PRODUCTS_QUANTITY_FIELD = "productsQuantity";
 
     private final ItemMarket itemMarket = new ItemMarket();
     private final Map<String, TreeSet<Item>> allItems = new ConcurrentHashMap<>();
+    private final Map<String, Long> productsQuantity = new ConcurrentHashMap<>();
 
-    @BeforeEach
+    @Before
     public void setUp() {
         TreeSet<Item> items = this.populateItems();
 
         allItems.put(EGGS_ITEM_NAME, items);
         allItems.put(RICE_ITEM_NAME, new TreeSet<>());
 
         ReflectionTestUtils.setField(itemMarket, ALL_ITEMS_FIELD, allItems);
+        ReflectionTestUtils.setField(itemMarket, PRODUCTS_QUANTITY_FIELD, productsQuantity);
+    }
+
+    @Test
+    public void updateItemUpdatesProductQuantityAndAllItemsMaps() {
+        //Arrange
+        itemMarket.addTrackedItem(CHEESE_ITEM_NAME);
+        TickResponse cheese = TickResponse.newBuilder().setGoodName(CHEESE_ITEM_NAME).setQuantity(2).setPrice(2.6).setOrigin(Origin.ASIA).build();
+        Aurora.AuroraResponse build = Aurora.AuroraResponse.newBuilder().setMessage(Any.pack(cheese)).build();
+
+        TickResponse cheese2 = TickResponse.newBuilder().setGoodName(CHEESE_ITEM_NAME).setQuantity(2).setPrice(2.6).setOrigin(Origin.ASIA).build();
+        Aurora.AuroraResponse build2 = Aurora.AuroraResponse.newBuilder().setMessage(Any.pack(cheese2)).build();
+
+        TickResponse cheese3 = TickResponse.newBuilder().setGoodName(CHEESE_ITEM_NAME).setQuantity(2).setPrice(2.6).setOrigin(Origin.AMERICA).build();
+        Aurora.AuroraResponse build3 = Aurora.AuroraResponse.newBuilder().setMessage(Any.pack(cheese3)).build();
+
+        //Act
+        itemMarket.updateItem(build);
+        itemMarket.updateItem(build2);
+        itemMarket.updateItem(build3);
+
+        //Assert
+        assertEquals(2, allItems.get(CHEESE_ITEM_NAME).size());
```

Comment:
a good check will be also to determine what is the quantity in the item itself, as we duplicate the information in products quantity, we need to be sure that the values match

Reply:
Fixed